gui/qt/external_plugins_window.py

```diff
@@ -0,0 +1,451 @@
+        self.checksumLabel.setText(_("Unavailable."))
+            
+        self.refresh_plugin(plugin_path)
+        self.refresh_ui()
+
+    def refresh_plugin(self, plugin_path):
+        self.plugin_path = plugin_path
+        
+        plugin_manager = self.main_window.gui_object.plugins
+        self.plugin_metadata = plugin_manager.get_metadata_from_external_plugin_zip_file(plugin_path)
+
+        hash_md5 = hashlib.md5()
+        with open(plugin_path, "rb") as f:
+            hash_md5.update(f.read())
+        self.checksumLabel.setText(hash_md5.hexdigest())
+        
+        self.pluginNameLabel.setText(self.plugin_metadata["display_name"])
+        if "version" in self.plugin_metadata:
+            self.versionLabel.setText(str(self.plugin_metadata["version"]))
+        self.descriptionLabel.setText(self.plugin_metadata["description"])
+        
+        available_for = self.plugin_metadata.get("available_for", [])
+        self.qtInterfaceCheckBox.setChecked("qt" in available_for)
+        self.cmdLineInterfaceCheckBox.setChecked("cmdline" in available_for)
+
+    def refresh_ui(self):
+        are_widgets_enabled = self.is_plugin_valid()
+        was_liability_accepted = self.liabilityCheckbox.checkState() == Qt.Checked
+
+        self.pluginNameLabel.setEnabled(are_widgets_enabled)
+        self.versionLabel.setEnabled(are_widgets_enabled)
+        self.descriptionLabel.setEnabled(are_widgets_enabled)
+        self.checksumLabel.setEnabled(are_widgets_enabled)
+        self.installButton.setEnabled(was_liability_accepted and are_widgets_enabled)
+        self.qtInterfaceCheckBox.setEnabled(are_widgets_enabled)
+        self.cmdLineInterfaceCheckBox.setEnabled(are_widgets_enabled)
+        self.liabilityCheckbox.setEnabled(are_widgets_enabled)
+
+    def is_plugin_valid(self):
+        return self.plugin_metadata is not None
+        
+    def on_liability_toggled(self):
+        self.refresh_ui()
+
+    def on_install(self):
+        self.close()
+
+        self.plugin_dialog.install_plugin_confirmed(self.plugin_path)
+
+
+class ExternalPluginsDialog(WindowModalDialog, MessageBoxMixin):
+    def __init__(self, parent, title):
+        WindowModalDialog.__init__(self, parent, title)
+
+        self.main_window = parent
+        self.config = parent.config
+        self.setMinimumWidth(600)
+        self.setMaximumWidth(600)
+
+        vbox = QVBoxLayout(self)
+
+        # The warning message box at the top of the dialog window about dangers of installing plugins.
+        self.descriptionGroupBox = QGroupBox()
+        self.descriptionGroupBox.setTitle(_("Security Warning"))
+        self.descriptionGroupBox.setAlignment(Qt.AlignHCenter)
+        descriptionGroupLayout = QVBoxLayout()
+        self.descriptionGroupBox.setLayout(descriptionGroupLayout)
+        self.descriptionLabel = QLabel(_("Install plugins at your own risk.\nThey have almost complete access to Electron Cash's internals."))
```

Comment:
I kind of feel like maybe one or two obnoxious popup warnings would be nice at some point in the installation process. Well, not nice, but you know what I mean.

Reply:
For me this has a nice balance.  You're going to have to fight that battle on your own :-)